umpost/um2netcdf.py

```diff
@@ -414,6 +365,65 @@ def cubewrite(cube, sman, compression, use64bit, verbose):
         sman.write(cube, zlib=True, complevel=compression, fill_value=fill_value)
 
 
+def fix_forecast_reference_time(cube):
+    # If reference date is before 1600 use proleptic gregorian
+    # calendar and change units from hours to days
+    try:
+        reftime = cube.coord(FORECAST_REFERENCE_TIME)
+        time = cube.coord(TIME)
+        refdate = reftime.units.num2date(reftime.points[0])
+        assert time.units.origin == 'hours since 1970-01-01 00:00:00'
+
+        if time.units.calendar == cf_units.CALENDAR_PROLEPTIC_GREGORIAN and refdate.year < 1600:
+            convert_proleptic(time)
+        else:
+            if time.units.calendar == cf_units.CALENDAR_GREGORIAN:
+                new_calendar = cf_units.CALENDAR_PROLEPTIC_GREGORIAN
+            else:
+                new_calendar = time.units.calendar
+
+            time.units = cf_units.Unit("days since 1970-01-01 00:00", calendar=new_calendar)
+            time.points = time.points / 24.
+
+            if time.bounds is not None:
```

Comment:
I wonder if it wouldn't be better to just check if `cube` has a FORECAST_REFERENCE_TIME coord outside this routine and only call it if so?

Then you could dump the `try`/`except` block.

As it is removing `FORECAST_PERIOD` too then I guess that either needs to be checked for or put in a smaller `try`/`except`

Reply:
> As it is removing `FORECAST_PERIOD` too then I guess that either needs to be checked for or put in a smaller `try`/`except`

The `iris` function `remove_coord()` does a `coord()` lookup, which raises an exception if the forecast period is missing:
https://github.com/SciTools/iris/blob/cc60a60508b741709ace45b8ea4fc05e5cd5e8a6/lib/iris/cube.py#L1813-L1830

I think we should consider swapping the 2 `remove_coord()` calls. Remove the forecast reference time first, as it's from to exist from the `try` test. then attempt to remove the forecast period. The current code could fail on removing the forecast period, leaving behind the forecast reference time, HOWEVER this might be the desired behaviour.